File: julia-mnp-plasmon/examples/spectrum_csv.jl
using DelimitedFiles
using MnpPlasmon

wavelengths = 400.0:5.0:700.0
rows = Vector{NTuple{4,Float64}}()
for wl in wavelengths
    r = sphere_response(wl, 25.0, "Au", 1.33)
    push!(rows, (wl, r.c_ext, r.c_sca, r.c_abs))
end

mkpath("docs/example-data")
open("docs/example-data/julia.csv", "w") do io
    write(io, "wavelength_nm,c_ext,c_sca,c_abs\n")
    for row in rows
        write(io, string(row[1], ",", row[2], ",", row[3], ",", row[4], "\n"))
    end
end

File: r-mnp-plasmon/examples/spectrum_csv.R
source("r-mnp-plasmon/R/mnp_plasmon.R")

wavelengths <- seq(400, 700, by = 5)
rows <- lapply(wavelengths, function(wl) {
  result <- sphere_response(
    wavelength_nm = wl,
    radius_nm = 25,
    material = "Au",
    medium_refractive_index = 1.33
  )
  data.frame(
    wavelength_nm = wl,
    c_ext = result$sigma_ext,
    c_sca = result$sigma_sca,
    c_abs = result$sigma_abs
  )
})

output <- do.call(rbind, rows)
dir.create("docs/example-data", recursive = TRUE, showWarnings = FALSE)
write.csv(output, file = "docs/example-data/r.csv", row.names = FALSE)
